scripts/check_doc_images.py

```
@@ -0,0 +1,104 @@
+            "Run: python scripts/sync_readme_gallery.py"
+        ]
+    return []
+
+
+def find_duplicate_images() -> list[str]:
+    shots = ROOT / "docs" / "screenshots"
+    if not shots.exists():
+        return []
+    by_hash: dict[str, list[str]] = {}
+    for png in shots.rglob("*.png"):
+        digest = hashlib.md5(png.read_bytes()).hexdigest()
```

Ruff (0.15.18)

[error] Probable use of insecure hash functions in `hashlib`: `md5`

(S324)